abbyy_to_epub3/image_processing.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2017 Deborah Kaplan
+#
+# This file is part of Abbyy-to-epub3.
+# Source code is available at <https://github.com/deborahgu/abbyy-to-epub3>.
+#
+# Abbyy-to-epub3 is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from PIL import Image
+
+import logging
+import subprocess
+
+
+class ImageProcessor(object):
+    """
+    The image object.
+
+    Can use various image processing libraries via factories.
+    """
+    def __init__(self, debug=False):
+        self.logger = logging.getLogger(__name__)
+        if debug:
+            self.logger.addHandler(logging.StreamHandler())
+            self.logger.setLevel(logging.DEBUG)
+
+
+def factory(type):
```

Comment:
We preferably don't want to clobber the `type` builtin

Reply:
I was surprised to find that this was the [Factory: Encapsulating Object Creation](https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Factory.html) recipe from Bruce Eckel's [Python 3 Patterns, Recipes and Idioms](https://python-3-patterns-idioms-test.readthedocs.io/en/latest/index.html)

 
https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Factory.html